akka-javasdk/src/main/java/akka/javasdk/client/ComponentMethodRef1.java

```diff
@@ -22,5 +23,21 @@ public interface ComponentMethodRef1<A1, R> extends ComponentDeferredMethodRef1<
 
   ComponentMethodRef1<A1, R> withMetadata(Metadata metadata);
 
+  /**
+   * Set the retry settings for this call.
+   *
+   * @param retrySettings The retry settings
+   * @return A new call with the retry settings set
+   */
+  ComponentMethodRef1<A1, R> withRetry(RetrySettings retrySettings);
+
+  /**
+   * Set the retry settings for this call. A predifined backoff strategy will be calculated based on the number of maxRetries.
+   *
+   * @param maxRetries The number of retries to make
+   * @return A new call with the retry settings set
+   */
+  ComponentMethodRef<R> withRetry(int maxRetries);
```

Comment:
this should return ComponentInvokeOnlyMethodRef1
btw, ComponentMethodRef is wrong, it should have been ComponentMethodRef1 here?